Set Model parameter distributions and load statements using DataFrame or spreadhseet

sarangbhagwat opened the pull request:
Adds a new EasyInputModel class that lets users set parameter distributions and load statements using DataFrame objects or spreadsheet files.

yoelcortes commented:
@sarangbhagwat,

This looks like a fun way to create models. I hope we can work it into the main code a little more. I would suggest the following edits:

    Remove the subclass, add the new method to the Model class, rename the new method to parameters_from_df.
    Remove namespace_dict from the model __init__, leave it as an input to the parameters_from_df method and rename it to namespace (considering python namespaces are dictionaries, it think we can opt for a shorter name).
    Replace namespace default to None (using mutable objects may lead to issues later).
    Add new feature to autofill the namespace with all objects in the flowsheet (except if name is already taken):

if namespace is None: 
    namespace = self.system.flowsheet.to_dict()
else:
    namespace = self.system.flowsheet.to_dict() | namespace

    Add a test in tests/test_evaluation.py

Thank you! Let me know your thoughts or if you'd like me to help with any implementation.

sarangbhagwat commented:
Yeah! Pretty much all the recent BioSTEAM biorefineries (TAL, HP, succinic) are using this to load parameter distributions, it's really useful close to publication when you can essentially just copy the table from your spreadsheet and paste it into your supporting information, haha. I'd implemented this feature in BioIndustrial-Park, just moving it to BioSTEAM.

Thanks, all great suggestions -- I've changed the code for the namespace dict merge slightly, let me know if you would prefer it the other way. Also added tests -- I've tried to cover most of the new features.

yoelcortes reviewed (approved):
This is awesome, now we have a clean way to define parameters in a spreadsheet. I appreciate you taking the time to move this feature from your biorefineries to core biosteam and addressing everything in the review. 

By the way, I removed the left over EasyInputModel subclass since its main feature was already added to the Model class. 

If you'd like to add aliases for backwards compatibility (avoiding duplicate code); I would be OK with that. For example, adding the following lines is OK.

```python
EasyInputModel = Model
Model.load_parameter_distributions = Model.parameters_from_df
```

Feel free to merge whenever,
Thank you!
